### PR TITLE
[tesseract] update to 5.0.1

### LINKED ISSUE #16019: [tesseract] update to 5.x.x
Library name: tesseract

New version number: 5.0.0-alpha-20201231

https://github.com/tesseract-ocr/tesseract/releases/tag/5.0.0-alpha-20201231

https://abi-laboratory.pro/?view=timeline&l=tesseract
5.0.0-alpha is only 36% backward compatible.
So vcpkg should have options to build the old 4.1.1 as well as new 5.0.0-alpha (with HEAD_REF master).

Opening a new issue as suggested by @NancyLi1013 in https://github.com/microsoft/vcpkg/issues/10624#event-4283543880

### LINKED ISSUE #24289: [tesseract[training-tools]] build failure
**Host Environment**
- OS: Windows 10
- Compiler: MSVC17

**To Reproduce**
`vcpkg install tesseract[cpu-independed,training-tools]:x64-windows`

**Failure logs**

```
-- Restarting build without parallelism
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:158 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j17
    Working Directory: C:/vcpkg/buildtrees/tesseract/x64-windows-dbg
    See logs for more information:
      C:\vcpkg\buildtrees\tesseract\install-x64-windows-dbg-out.log
      C:\vcpkg\buildtrees\tesseract\install-x64-windows-dbg-out-1.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:111 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:41 (vcpkg_cmake_build)
  ports/tesseract/portfile.cmake:39 (vcpkg_cmake_install)
  scripts/ports.cmake:145 (include)


error: building tesseract:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `git pull` and `.\vcpkg update`.
Then check for known issues at:
  https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+tesseract
You can submit a new issue at:
  https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[tesseract]+Build+error
including:
  package: tesseract[core,training-tools]:x64-windows -> 4.1.1#10
    vcpkg-tool version: 2022-03-30-692785ac944e81417840c6de244fb3e18a4b35eb
    vcpkg-scripts version: e0c3408c7 2022-04-20 (10 hours ago)

Additionally, attach any relevant sections from the log files above.
```

**Additional context**
How do you solve this, forking the portfiles did not resolve my issue.

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #16019

Update tesseract to the latest version 5.0.1

All features are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

### CODE REVIEW

File: ports/tesseract/fix-findpackage.patch [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+-if(WIN32)
+-  option(SW_BUILD "Build with sw" ON)
+-else()
+-  option(SW_BUILD "Build with sw" OFF)
+-endif()
++
++option(SW_BUILD "Build with sw" OFF)
+ option(OPENMP_BUILD "Build with openmp support" OFF) # see issue #1662
+ option(GRAPHICS_DISABLED "Disable disable graphics (ScrollView)" OFF)
+ option(DISABLED_LEGACY_ENGINE "Disable the legacy OCR engine" OFF)
```
Comment (cenit) [inline ports/tesseract/fix-findpackage.patch:18]: This patch is totally unnecessary, better to pass -DSW_BUILD=OFF inside cmake configure step!

Comment (cenit) [reply ports/tesseract/fix-findpackage.patch:18]: but with this change, you are removing tools that were available for downstream usage!

Comment (JonLiu1993) [reply ports/tesseract/fix-findpackage.patch:18]: @cenit ,Good idea!

**Comment by Shatur** [comment]
@JonLiu1993 are you planning to continue working on it?

**Comment by JonLiu1993** [comment]
> @JonLiu1993 are you planning to continue working on it?

Yeah, I will fix this pr asap

**Comment by Shreeshrii** [comment]
@egorpugin please look at the mentioned issue related to sw config. Thanks!

**Comment by Shatur** [comment]
@JonLiu1993 are you planning to continue it?

**Comment by JonLiu1993** [comment]
> @JonLiu1993 are you planning to continue it?

@Shatur，I will reopen and update pr after I fix this locally

**Pull request closed without merging**